syntax="proto3";

package sftp;
option go_package="sftppb";
option java_multiple_files = true;
option java_package = "io.grpc.examples.sftp";
option java_outer_classname = "SFTP";
option objc_class_prefix = "SFTP";

message Sftp {
	string username = 1;
	string passWord = 2;
	string systemId = 3;
	string hostKey  = 4;
	string fileName = 5;
	string hostPort = 6;
	string clientKey = 7;
}

message CopyLocalToRemoteRequest{
	Sftp sftp = 1;
}
message CopyLocalToRemoteResponse{
	string result = 1;
}
message CopyFromRemoteRequest{
	Sftp sftp = 1;
}
message CopyFromRemoteResponse{
	string result = 1;
}
service SFTP{
	rpc CopyFromRemoteService(CopyFromRemoteRequest) returns (CopyFromRemoteResponse);
	rpc CopyLocalToRemoteService(CopyLocalToRemoteRequest) returns (CopyLocalToRemoteResponse);
}

